packages/chart/src/Pie.ts

```diff
@@ -155,62 +203,55 @@ export class Pie extends SVGWidget {
-                    const mid_angle = midAngle(d2);
-                    pos[0] = labelRadius * (mid_angle < Math.PI && mid_angle > 0 ? 1 : -1);
-                    return "translate(" + pos + ")";
-                };
-            })
-            .styleTween("text-anchor", function (d) {
-                this._current = this._current || d;
-                const interpolate = d3Interpolate(this._current, d);
-                this._current = interpolate(0);
-                return function (t) {
-                    const d2 = interpolate(t);
-                    const mid_angle = midAngle(d2);
-                    return mid_angle < Math.PI && mid_angle > 0 ? "start" : "end";
-                };
+            .text(d => this.getLabelText(d, true))
+            .transition()
+            .each(function (d, i) {
+                const pos = context.d3LabelArc.centroid(d);
+                const mid_angle = midAngle(d);
+                pos[0] = labelRadius * (context.isLeftSide(mid_angle) ? 1 : -1);
+                context._labelPositions.push({
+                    top: pos[1],
+                    bottom: pos[1] + context.labelHeight()
+                });
             });
-
-        function midAngle(d) {
-            return d.startAngle + (d.endAngle - d.startAngle) / 2;
-        }
+        this.adjustForOverlap();
+        mergedText
+            .style("font-size", this.labelHeight() + "px")
+            .attr("transform", (d, i) => {
+                const pos = context.d3LabelArc.centroid(d);
+                pos[0] = labelRadius * (context.isLeftSide(midAngle(d)) ? 1 : -1);
+                pos[1] = context._labelPositions[i].top;
+                return "translate(" + pos + ")";
+            })
+            .transition()
+            .style("text-anchor", d => this.isLeftSide(midAngle(d)) ? "start" : "end");
 
         text.exit()
             .remove();
 
-        const polyline = this._labels.selectAll("polyline").data(this.d3Pie(this.data()), d => this.getLabelText(d));
+        const polyline = this._labels.selectAll("polyline").data(this.d3Pie(this.data()), d => this.getLabelText(d, true));
 
         polyline.enter()
             .append("polyline")
             .merge(polyline)
-            .transition().duration(1000)
-            .attrTween("points", function (d) {
-                this._current = this._current || d;
-                const interpolate = d3Interpolate(this._current, d);
-                this._current = interpolate(0);
-                return function (t) {
-                    const d2 = interpolate(t);
-                    const pos = context.d3LabelArc.centroid(d2);
-                    const pos1 = context.d3Arc.centroid(d2);
-                    const pos2 = [...pos];
-                    const mid_angle = midAngle(d2);
-                    pos[0] = labelRadius * (mid_angle < Math.PI && mid_angle > 0 ? 1 : -1);
-                    return [pos1, pos2, pos];
-                };
+            .transition()
```

Comment:
By convention I typically put the transition on the same line as the previous call:  https://github.com/hpcc-systems/Visualization/blob/master/packages/chart/src/Pie.ts#L119